tests/unit_tests/test_config.py

```
@@ -96,3 +96,16 @@ def test_override_mps(monkeypatch, tiny_config, tmp_path, caplog):
             "overwritten to 'cpu' on Apple Silicon" in rec.getMessage()
             for rec in caplog.records
         )
+
+
+def test_invalid_train_batch_size(tmp_path, tiny_config):
+    """Test that a non-positive train_batch_size raises a ValueError."""
+    for bad_value in [0, -1]:
+        filename = str(tmp_path / f"config_bad_batch_{bad_value}.yml")
+        with open(tiny_config, "r") as f_in, open(filename, "w") as f_out:
```

Ruff (0.15.7)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)